staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -54,28 +54,74 @@ var (
 		kubectl explain pods.spec.containers`))
 )
 
-type ExplainOptions struct {
+// ExplainFlags directly reflect the information that CLI is gathering via flags.
+// They will be converted to Options, which reflect the runtime requirements
+// for the command.
+type ExplainFlags struct {
+	Factory    cmdutil.Factory
+	APIVersion string
+	// Name of the template to use with the openapiv3 template renderer. If
+	// `EnableOpenAPIV3` is disabled, this does nothing
+	OutputFormat string
+	Recursive    bool
 	genericclioptions.IOStreams
+}
 
-	CmdParent  string
-	APIVersion string
-	Recursive  bool
+// AddFlags registers flags for a cli
+func (flags *ExplainFlags) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&flags.Recursive, "recursive", flags.Recursive, "Print the fields of fields (Currently only 1 level deep)")
+	cmd.Flags().StringVar(&flags.APIVersion, "api-version", flags.APIVersion, "Get different explanations for particular API version (API group/version)")
+}
 
-	args []string
+// NewExplainFlags returns a default ExplainFlags
+func NewExplainFlags(f cmdutil.Factory, streams genericclioptions.IOStreams) *ExplainFlags {
+	return &ExplainFlags{
+		Factory:      f,
+		OutputFormat: "plaintext",
+		IOStreams:    streams,
+	}
+}
 
-	Mapper meta.RESTMapper
-	Schema openapi.Resources
+// ToOptions converts from CLI inputs to runtime input
+func (flags *ExplainFlags) ToOptions(cmd *cobra.Command, parent string, args []string) (*ExplainOptions, error) {
+	var err error
```

Comment:
No need to define error here, since you're using `:=` in the next line so it'll be already defined from then on.